routemaster/services/thread.rb

```diff
@@ -0,0 +1,62 @@
+require 'routemaster/services'
+require 'routemaster/mixins/log'
+require 'routemaster/mixins/log_exception'
+
+module Routemaster
+  module Services
+    # Abstracts out repeatedly calling `callable` on a thread.
+    # On exceptions, enqueues itself to `errq`.
+    #
+    # Long-running callables are expected to yield to check whether they may
+    # continue.
+    #
+    class Thread
+      include Mixins::Log
+      include Mixins::LogException
+
+      attr_reader :name
+      
+      def initialize(callable, name:, errq:)
+        @callable = callable
+        @name     = name
+        @errq     = errq
+        @running  = true
+        @thread   = ::Thread.new(&method(:_run))
+        @thread.abort_on_exception = true
+      end
+
+      def stop
+        _log.info { "stopping #{name}" }
+        @running = false
```

Comment:
mh ok got confused with the naming then, as I expected it to actually stop the thread not wait for it to finish whatever it is doing.